File: ChronosQL.Node/host.js
const { spawnSync } = require('child_process');
const path = require('path');

const sql = 'SELECT data.value FROM input';
const input = '{"data":{"value":1}}\n{"data":{"value":2}}';

const projectPath = path.join(__dirname, 'ChronosQL.Node.csproj');
const result = spawnSync('dotnet', ['run', '--project', projectPath, '--', sql, input], {
  encoding: 'utf8'
});

if (result.error) {
  console.error(result.error);
  process.exit(1);
}

if (result.stderr) {
  console.error(result.stderr.trim());
}

console.log(result.stdout.trim());

File: ChronosQL.Python/host.py
import os
import subprocess
import sys

sql = 'SELECT data.value FROM input'
input_payload = '{"data":{"value":1}}\n{"data":{"value":2}}'
project_path = os.path.join(os.path.dirname(__file__), 'ChronosQL.Python.csproj')

result = subprocess.run(
    ['dotnet', 'run', '--project', project_path, '--', sql, input_payload],
    capture_output=True,
    text=True,
)

if result.returncode != 0:
    sys.stderr.write(result.stderr)
    sys.exit(result.returncode)

if result.stderr:
    sys.stderr.write(result.stderr)

print(result.stdout.strip())
